app/Models/Torrent.php

```diff
@@ -463,4 +467,32 @@ public function isFreeleech($user = null)
 
         return $this->free || \config('other.freeleech') || $pfree;
     }
+
+    /**
+     * Torrent Is Promo 50%.
+     *
+     * @param null $user
+     *
+     * @return bool
+     */
+    public function isPromo50($user = null)
+    {
+        $pPromo50 = $user ? $user->group->is_freeleech || PersonalFreeleech::where('user_id', '=', $user->id)->first() : false;
+
+        return $this->free || \config('other.freeleech') || $pfree;
+    }
+
+    /**
+     * Torrent Is Promo 25%.
+     *
+     * @param null $user
+     *
+     * @return bool
+     */
+    public function isPromo25($user = null)
```

Comment:
`$user = null` is not needed here. `$pPromo25 = $user ? $user->group->is_freeleech || PersonalFreeleech::where('user_id', '=', $user->id)->first() : false;` can be removed as well. `return $this->free || \config('other.freeleech') || $pfree;` should just be `return $this->promo25`